src/tools/prefs.mac.mm

```diff
@@ -94,11 +106,65 @@ void append_iarray_pref(std::string keypath, int value) {
 }
 
 void clear_pref(std::string keypath) {
-	[[NSUserDefaults standardUserDefaults] setValue: nil forKey: convertKey(keypath)];
+	[getCurrentDefaults() setValue: nil forKey: convertKey(keypath)];
+}
+
+// When replaying, load the preferences from the action log into a
+// non-synchronized UserDefaults object
+static bool load_prefs(std::istream& istream) {
+	std::string line;
+	while(std::getline(istream, line)) {
+		if(!istream) {
+			perror("Error reading preferences");
+			return false;
+		}
+		if(line[0] == '#') continue;
+		int eq = line.find_first_of('=');
+		if(eq == std::string::npos) {
+			printf("Error reading preferences: line is not a comment and lacks an = sign:\n\t%s\n", line.c_str());
```

Comment:
Is there some reason why we can't use streaming like the rest of the code?

Reply:
I started by copying over the implementation of load_prefs() for windows/linux:

https://github.com/calref/cboe/blob/10ad8dc9da4a73eb4ad2895b3d1dcc104c48f68a/src/tools/prefs.win.cpp#L105

I wouldn't want to change too much from something we're pretty sure works fine.

Reply:
Well, I guess we can leave it as is for now; I'd probably want the Linux version to also use streaming there but keeping them as much the same as possible isn't a bad goal.